### PR TITLE
Added framing test for setFocalPlane.  

### PR DESCRIPTION
**Pull request by twilson271828**
I also added the ability to test protected/private member functions in classes using the FRIEND_TEST macro in gtest (although I am not happy with this particular implementation).

Additionally in the test fixture, I load the ISD and replace the odt_x,odt_y vectors with those defined within
the ik kernel for the MDIS NAC.  I did this through the ISD.  There might be a better way to do it
through the nlohmann::json api though.

### CODE REVIEW

**Comment by jlaura** [comment]
Second to not using real values. If we end up with a corner case that requires real (or pseudo-real) values, we should take a long look at why this is the case. At this point, we need to be doing tests on paper using simple values to ensure that we are 100% correct. The problem with real values is that we are (hopefully) going to start to build on top of these methods and we need them to be bomber.

**Comment by twilson271828** [comment]
I'll try that (modifying the ISD fixture).  The only way to test actual distortion in this case would be to use an actual MDIS NAC image.

**Comment by jessemapel** [comment]
@twilson271828 We can use the test "image" and introduce artificial distortion. That will test without a real image

**Comment by twilson271828** [comment]
So any comments about the way I modified the testing to test protected/private class members?

**Comment by jessemapel** [comment]
@twilson271828 The private member testing looks good. The gtest docs recommend not using the private testing tools and refactor, but because of the CSM API we can't. So, I think this is the best way to test those types of functions.

File: tests/FrameCameraTests.cpp [hunk 1]
```diff
@@ -101,3 +129,32 @@ TEST_F(FrameSensorModel, OffBody4) {
...
+      strval << setprecision(precision) << val;
+      localIsd.addParam("odt_x", strval.str());
+   }
+   for (auto & val: odty){
+      ostringstream strval;
+      strval << setprecision(precision) << val;
+      localIsd.addParam("odt_y", strval.str());
+   }
+
+   sensorModel->setFocalPlane(imagePt.samp, imagePt.line, ux, uy);
```
Comment (jessemapel) [inline tests/FrameCameraTests.cpp]: The sensor model is not updated from your local ISD.  You need to modify the ISD and then generate a camera model from it. Look at how @AstroKEW78 wrote the focal length test https://github.com/USGS-Astrogeology/CSM-CameraModel/blob/master/tests/FrameCameraTests.cpp#L127

File: tests/FrameCameraTests.cpp [hunk 2]
```diff
@@ -124,20 +147,51 @@ TEST_F(FrameSensorModel, OffBody4) {
...
+   EXPECT_NEAR(ux,7.0,1e-8 );
+   EXPECT_NEAR(uy,7.0,1e-8);
```
Comment (jessemapel) [inline tests/FrameCameraTests.cpp]: Shouldn't these by 14,63 ?

File: tests/FrameCameraTests.cpp [hunk 3]
```diff
@@ -124,20 +147,51 @@ TEST_F(FrameSensorModel, OffBody4) {
...
 
+TEST_F(FrameSensorModel, setFocalPlane1) {
+  int precision  = 20;
+
+   csm::Isd localIsd;
+   loadIsd("data/simpleFramerISD.json",localIsd);
+   csm::ImageCoord imagePt(7.0, 7.0);
+   double ux,uy;
+   vector<double> odtx{0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+   vector<double> odty{0.0,1.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
```
Comment (jessemapel) [inline tests/FrameCameraTests.cpp]: All 0 vectors should be no distortion. This should just change one coefficient for x and one coefficient for y. Maybe just one coefficient at a time. This thing can be super parameterized.

File: tests/FrameCameraTests.cpp [hunk 4]
```diff
@@ -124,20 +143,201 @@ TEST_F(FrameSensorModel, OffBody4) {
...
+   }
+
+   csm::Model *model = frameCameraPlugin.constructModelFromISD(
+         isd,
+         "USGS_ASTRO_FRAME_SENSOR_MODEL");
+
+   UsgsAstroFrameSensorModel* sensorModel = dynamic_cast<UsgsAstroFrameSensorModel *>(model);
+
+
+   printIsd(isd);
```
Comment (jessemapel) [inline tests/FrameCameraTests.cpp]: I can see this is helpful for debugging but I don't think we want to actually have this called as part of our test.
